lib/adapters/leveldb/index.js

```diff
@@ -10,7 +10,16 @@ function requireLeveldown() {
   // de-optimalization
   try {
     originalLeveldown = require('leveldown');
-  } catch (e) {}
+  } catch (err) {
+      if (err && err.message && err.message.match('Module version mismatch')) {
+          throw new Error([
+            err.message,
+            'This generally implies that leveldown was built with a different',
+            'version of node than that which is running now.  You may try',
+            'fully removing and reinstalling PouchDB or leveldown to resolve.'
+          ].join(' '));
+      }
```

Comment:
[Check this out](https://github.com/nodejs/node/blob/master/lib/module.js#L338).  Module not found is part of nodejs core, and can be probably be relied on.  It's [been there since 2011](https://github.com/nodejs/node/commit/3f987cd925dd342b4695afd9c221bc19e76bdf7b), and I wouldn't forsee it changing.  Also, ["Module version mismatch"](https://github.com/nodejs/node/blob/master/src/node.cc#L2128) is also part of node core.  It doesn't have a standard ERROR code, which is a bummer, but is also stable over the past 5 years.

Although grateful for the support :), I think putting my Error where the current error block is not a good idea.  This is because there are a variety of _other_ errors that it could happen while doing a `require('leveldown')`, and I'd hate to misdirect users by stating their problem is possibly related to a build problem if the root cause is something else entirely.  For instance, perhaps there's an `EACCES` error because someone has fubared their node_modules permissions, etc.  When I learned node, I did that all of the time :)

With this in mind, how would you feel about the follow up commit?  Test's re-ran, using async CB only now.

```js
// example.js
var P = require('./'); // ran in pouchdb root
var p = new P('dbname');
p.get('abcd').catch(function(err) {
    console.log(err.message);
   // => mismatch error if leveldown built on different node version OR,
   // 'Error: EACCES, permission denied ... ' if theres a perm error, for example, OR,
   // "the 'leveldown' package is not available. install it, or, specify another storage backend using the 'db' option" if leveldown missing
   // i tested each case
});
setInterval(function() {}, 1000);
```

finally, [optional](https://github.com/tony-o/node-optional/blob/master/optional.js) offers these features, but it doesn't have a great API and doesnt have test cases.  this whole thought process may inspire me to write a better version of that